holoviews/element/raster.py

```diff
@@ -259,6 +260,18 @@ def __init__(self, data, bounds=None, extents=None, xdensity=None, ydensity=None
                 raise ValueError("Input array has shape %r but %d value dimensions defined"
                                  % (self.shape, len(self.vdims)))
 
+
+    def __setstate__(self, state):
+        """
+        Ensures old-style unpickled Image types without an interface
+        use the ImageInterface.
+        """
```

Comment:
Right, in fact we really need to remove the ``__setstate__`` on Dataset as that's for the old terrible NdElement format, which was deprecated in 1.4(?) iirc and might be the only reason this is required at all.